packages/cli/templates/vue3/src/utils/convertToRaw.ts

```diff
@@ -0,0 +1,24 @@
+import { toRaw, isProxy, isRef, unref } from 'vue'
+
+export function convertToRaw(value: any) {
+  let rawValue = value
+  if (isProxy(rawValue)) {
+    rawValue = toRaw(rawValue)
+  }
+  if (isRef(rawValue)) {
+    rawValue = unref(rawValue)
+  }
+
+  if (!isObject(rawValue)) {
+    return rawValue
+  }
+
+  return Object.keys(rawValue).reduce((result, key) => {
+    result[key] = convertToRaw(rawValue[key])
+    return result
+  }, rawValue)
+}
+
+function isObject(value: unknown) {
+  return Object.prototype.toString.call(value) === '[object Object]'
+}
```

Comment:
Would this not work: 

```
typeof value === 'object' && value !== null && !Array.isArray(value)
```

Reply:
We should avoid other types of object like `new Date()`.